tools/gen_usb_descriptor.py

```diff
@@ -8,6 +8,15 @@
 from adafruit_usb_descriptor import audio, audio10, cdc, hid, midi, msc, standard, util
 import hid_report_descriptors
 
+ALL_DEVICES='CDC,MSC,AUDIO,HID'
+ALL_DEVICES_SET=frozenset(ALL_DEVICES.split(','))
+DEFAULT_DEVICES='CDC,MSC,AUDIO,HID'
+
+ALL_HID_DEVICES='KEYBOARD,MOUSE,CONSUMER,SYS_CONTROL,GAMEPAD,DIGITIZER,XAC_COMPATIBLE_GAMEPAD,RAW'
+ALL_HID_DEVICES_SET=frozenset(ALL_HID_DEVICES.split(','))
+# Digitizer works on Linux but conflicts with mouse, so omit it.
+DEFAULT_HID_DEVICES='KEYBOARD,MOUSE,CONSUMER,GAMEPAD'
+
```

Comment:
My old PR had a USB_CDC_AND_MSC_ONLY = 1 flag here, which was then used to reduce the MSC endpoints. The MSC endpoints only need to be reduced for the F411 and other low-endpoint STM32 boards, but reducing the endpoints will only fail on the SAMD21. Could we put in either a new flag like MSC_SINGLE_ENDPOINT, or have the MSC endpoints both reduced to 0 if MIDI and AUDIO are both turned off?

Reply:
Alternatively, we could do it the other way and have a "MSC_DUAL_ENDPOINT" flag that only forces two MSC endpoints for the SAMD21.

Reply:
I added a `USB_MSC_NUM_ENDPOINT_PAIRS` makefile variable and corresponding arg in `gen_usb_descriptor.py`. It defaults to 1 and is set to 2 for SAMD21 boards.

Reply:
I tested this on M0, M4, and nRF boards, and checked the generated descriptor as well to make sure it was choosing right.

It is indeed the case still that SAMD21 needs a separate endpoint pair. Otherwise it will sometimes enumerate (sometimes works after a reset) but sometimes not (if plugged in fresh).